---
title: ""
author: ""
date: ""
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    number_sections: false
    toc_depth: 3
    smooth_scroll: false
    theme: cerulean
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reticulate)
library(knitr)
library(ChIPseeker)
library(AnnotationDbi)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(org.Mm.eg.db)
library(dplyr)
library(reactable)
library(data.table)
txdb <- TxDb.Mmusculus.UCSC.mm10.knownGene
options(ChIPseeker.ignore_1st_exon = TRUE)
options(ChIPseeker.ignore_1st_intron = TRUE)
options(ChIPseeker.ignore_downstream = TRUE)
options(ChIPseeker.ignore_promoter_subcategory = TRUE)
setwd("~/Documents/K36me2_enhancer")
```

# Number of peaks per bed - semi-supervised 
```{python include = FALSE}
import pandas as pd
df = pd.DataFrame({"Type" : ["MACS3","SPAN before merging","SPAN after merging"],
                  "# of Peaks" : [67497,39068,11944]
})
print(df.to_markdown(index = False))

#about 13000 less peaks in 0.01 but many of these are small peaks
#The IDR algorithm requires sampling of both signal and noise distributions to separate the peaks into two groups, so having a more liberal threshold allows us to bring in some noise.
```

```{r echo = FALSE}
kable(py$df)
```

# Peaks annotation

```{r include = FALSE}
#load data
samplefiles <- list.files("~/Documents/K36me2_enhancer", pattern= "merged_final.bed", full.names=T)
samplefiles <- as.list(samplefiles)
names(samplefiles) <- c("TKO_K36me2")
peakAnnolist <- lapply(samplefiles,annotatePeak,TxDb = txdb,tssRegion = c(-1000, 1000))
plotAnnoBar(peakAnnolist)
annot_df <- peakAnnolist[["TKO_K36me2"]]@anno %>% as.data.frame()
```

```{r include = FALSE}
peakAnnolist[["TKO_K36me2"]]@annoStat %>% reactable()
```

Most of the peaks are found in promoters (35%) and distal intergenic regions (26%).
```{r include = FALSE}
peakAnnolist_V2 <- lapply(samplefiles,annotatePeak,TxDb = txdb,tssRegion = c(-1000, 1000))
```

```{r echo = FALSE,warning=FALSE,message=FALSE,out.width='100%'}
plotAnnoBar(peakAnnolist_V2)
peakAnnolist_V2[["TKO_K36me2"]]@annoStat %>% reactable()
```

# Distance to TSS

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plotDistToTSS(peakAnnolist_V2)
```

```{r include = FALSE}
enhancer_bed <- fread("~/Documents/K36me2_enhancer/K36me2.enhancer.bed") %>%
  setNames(c("chr","chrStart","chrEnd","peak_id","hits"))
#number of peaks with at least 1 hit in enhancer.bed (at least one enhancer overlaps with peak)
promoter_bed <- fread("~/Documents/K36me2_enhancer/K36me2.promoter.bed") %>%
  setNames(c("chr","chrStart","chrEnd","peak_id","hits"))
intergenic_bed <- fread("~/Documents/K36me2_enhancer/K36me2.intergenic.bed") %>%
  setNames(c("chr","chrStart","chrEnd","peak_id","hits"))
exon_bed <- fread("~/Documents/K36me2_enhancer/K36me2.exon.bed") %>%
  setNames(c("chr","chrStart","chrEnd","peak_id","hits"))
intron_bed <- fread("~/Documents/K36me2_enhancer/K36me2.intron.bed") %>%
  setNames(c("chr","chrStart","chrEnd","peak_id","hits"))
####
original_enhancer_bed <- fread("~/Documents/K36me2_enhancer/enhancer.bed")
merged_bed <- fread("merged_final.bed")
###################
count_annot <- function(x){
  count <- (nrow(x[x$hits >= 1,])/nrow(x))*100
  return(count)
}
# Minimum overlap required as a fraction of B. Default is 1E-9 (i.e., 1bp). genomic region must be 100% within the peak to be considered a hit
# feature in genomic region overlaps at least 100% of the peaks 
#candidate cis-Regulatory Elements (cCREs) 

#hypothesis: between these double peaks are enhancer elements 
#assumptions: from an analysis point of view: peak calling is done right, based on previously described enhancer elements
#from an experiment point of view: quality of antibody, lacking replicates

#TO DO
#classify enhancer as close to promoter
#not close to promoter or gene
#enhancer in gene body or transcription start site

```

# Promoter {.tabset}

## Plot
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.promoter.profile.png",dpi=600)
include_graphics("images/K36me3.promoter.profile.png",dpi=600)
include_graphics("images/K27ac.promoter.profile.png",dpi=600)
```

## Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.promoter.heatmap.png",dpi=600)
include_graphics("images/K36me3.promoter.heatmap.png",dpi=600)
include_graphics("images/K27ac.promoter.heatmap.png",dpi=600)
```

# Enhancer {.tabset}

## Plot
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.enhancer.profile.png",dpi=600)
include_graphics("images/K36me3.enhancer.profile.png",dpi=600)
include_graphics("images/K27ac.enhancer.profile.png",dpi=600)
```

## Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.enhancer.heatmap.png",dpi=600)
include_graphics("images/K36me3.enhancer.heatmap.png",dpi=600)
include_graphics("images/K27ac.enhancer.heatmap.png",dpi=600)
```

# TSS {.tabset}

## Plot
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.TSS.profile.png",dpi=600)
include_graphics("images/K36me3.TSS.profile.png",dpi=600)
include_graphics("images/K27ac.TSS.profile.png",dpi=600)
#Using that matrix we can create a profile plot which is essentially a density plot that evaluates read density across all transcription start sites. We can see that TKO has a particularly higher amount of signal at the TSS compared to other samples.
```

## Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2.TSS.heatmap.png",dpi=600)
include_graphics("images/K36me3.TSS.heatmap.png",dpi=600)
include_graphics("images/K27ac.TSS.heatmap.png",dpi=600)
```

# ATACseq

```{r include=FALSE}
#load data
samplefiles <- list.files("~/Documents/K36me2_enhancer/ATACseq", pattern= "_peaks.bed", full.names=T)
samplefiles <- as.list(samplefiles)
names(samplefiles) <- c("K36M_ATAC","NSD12DKO_ATAC","PA_ATAC","SETD2KO_ATAC")
peakAnnolist <- lapply(samplefiles,annotatePeak,TxDb = txdb,tssRegion = c(-1000, 1000))
#plotAnnoBar(peakAnnolist)
# rearrange 
#peakAnnolist[["SETD2KO_ATAC"]]@anno %>% as.data.frame() %>% write.csv(file = "~/Documents/K36me2_enhancer/ATACseq/setd2ko_atac.csv")
#invisible(lapply(names(peakAnnolist),function(x) assign(x,peakAnnolist[[x]] %>% as.data.frame() %>% write.csv(paste0(x,".csv")),.GlobalEnv)))
```

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plotAnnoBar(peakAnnolist)
```

## ATACseq {.tabset}

### Plots
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/ATACseq_ATAC.promoter.profile.png",dpi=600)
include_graphics("images/ATACseq_ATAC.intron.profile.png",dpi=600)
include_graphics("images/ATACseq_ATAC.intergenic.profile.png",dpi=600)
```

### Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/ATACseq_ATAC.promoter.heatmap.png",dpi=600)
include_graphics("images/ATACseq_ATAC.intron.heatmap.png",dpi=600)
include_graphics("images/ATACseq_ATAC.intergenic.heatmap.png",dpi=600)
```

## K27ac {.tabset}

### Plots
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K27ac_ATAC.promoter.profile.png",dpi=600)
include_graphics("images/K27ac_ATAC.intron.profile.png",dpi=600)
include_graphics("images/K27ac_ATAC.intergenic.profile.png",dpi=600)
```

### Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K27ac_ATAC.promoter.heatmap.png",dpi=600)
include_graphics("images/K27ac_ATAC.intron.heatmap.png",dpi=600)
include_graphics("images/K27ac_ATAC.intergenic.heatmap.png",dpi=600)
```

## K36me2 {.tabset}

### Plots

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2_ATAC.promoter.profile.png",dpi=600)
include_graphics("images/K36me2_ATAC.intron.profile.png",dpi=600)
include_graphics("images/K36me2_ATAC.intergenic.profile.png",dpi=600)
```

### Heatmap
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', dpi = 600,fig.retina=2}
include_graphics("images/K36me2_ATAC.promoter.heatmap.png",dpi=600)
include_graphics("images/K36me2_ATAC.intron.heatmap.png",dpi=600)
include_graphics("images/K36me2_ATAC.intergenic.heatmap.png",dpi=600)
```

## K4me1

## K4me3

